core/src/main/scala/chisel3/Bits.scala

```diff
@@ -148,6 +148,21 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
 
   /** @group SourceInfoTransformMacro */
   final def do_extract(x: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
+    this.widthOption.foreach { thisWidth =>
+      if (thisWidth == 0) {
+        Builder.error(s"Cannot extract from zero-width")
```

Comment:
```suggestion
        Builder.error(s"Cannot extract from zero-width $this")
```
We've improve `.toString` on `Data` to give more useful information, we should generally include it in error messages about specific `Data`.